### DISCUSSION TITLE
How to fetch new versions from folder in artifactory

### DISCUSSION THREAD

**Discussion by faenschi** [renovatebot/renovate#22285] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

HI all!

I have a Dockerfile which has a version in an environent variable defined like that:

```
# renovate: datasource=artifactory depName=Proxy versioning=loose
ENV DEP_VERSION=23.5.0-java17
```
This dependency is found in the artifactory in a maven structure :

`https://x.y.z/ui/native/maven-releases-local/a/b/c/Proxy` and in this repo path all the subfolders with the versions are listed (maven structure):
```
23.2.0-java17
23.2.0-java11
23.8.0-java 17

```

how can i configure renovate to extract the versions from these folders? i tried something like that 👍 

```
{
        "description": "proxy",
        "matchDatasources": ["artifactory"],
        "registryUrls": ["`https://x.y.z/ui/native/maven-releases-local/a/b/c/"],
        "matchPackagePrefixes": ["Proxy"],
        "versioning": "loose",
        "extractVersion": "^(?<version>(\\d+)\\.(\\d+)\\.(\\d+)-java17)$"
      }
```
i always end up getting the following error:

```
TRACE: artifactory: No versions found (repository=renovate-playground)
       "registryUrl": "https://x.y.z/ui/native/maven-releases-local/a/b/c/v",
       "packageName": "Proxy"
DEBUG: Failed to look up artifactory package Proxy(repository=playground, packageFile=files/Dockerfile, dependency=Proxy)

```
any help appreciated
regards angela

### What you tried so far.

```
{
        "description": "proxy",
        "matchDatasources": ["artifactory"],
        "registryUrls": ["`https://x.y.z/ui/native/maven-releases-local/a/b/c/"],
        "matchPackagePrefixes": ["Proxy"],
        "versioning": "loose",
        "extractVersion": "^(?<version>(\\d+)\\.(\\d+)\\.(\\d+)-java17)$"
      }
```
I also tried with datasource maven.
### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
We don't currently have support for a generic or regex datasource, which is what you'd need
